app/main.py

```diff
@@ -1,88 +1,45 @@
-from typing import Hashable, Any
+from typing import Any
 
 
 class Dictionary:
     def __init__(self) -> None:
         self.length = 0
         self.capacity = 8
-        self.load_factor = 2 / 3
         self.hash_table: list = [None] * self.capacity
+        self.load_factor = 2 / 3
 
-    def __len__(self) -> int:
-        return self.length
-
-    def __setitem__(self, key: Hashable, value: Any) -> None:
-
-        index = self.get_index_for_key(key)
-
-        if self.hash_table[index] is not None:
-            for i, (existing_key, _, existing_value) \
-                    in enumerate(self.hash_table[index]):
-                if existing_key == key:
-                    self.hash_table[index][i] = (key, hash(key), value)
-                    break
-        else:
-            self.hash_table[index] = [(key, hash(key), value)]
-
-    def __getitem__(self, key: Hashable) -> Any:
-        index = self.get_index_for_key(key)
-        if self.hash_table[index] is None:
-            raise KeyError
-        return self.hash_table[index][2]
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.length >= self.capacity * self.load_factor:
+            self.resize()
+        key_index = self.get_index(key)
+        if not self.hash_table[key_index]:
+            self.length += 1
+        self.hash_table[key_index] = (key, hash(key), value)
+
+    def __getitem__(self, item: Any) -> Any:
+        item_index = self.get_index(item)
+        if self.hash_table[item_index]:
+            return self.hash_table[item_index][2]
+        raise KeyError
 
-    def get_index_for_key(self, key: Hashable) -> int:
+    def get_index(self, key: Any) -> int:
```

Comment:
key must be Hashable

Reply:
![image](https://github.com/mate-academy/py-dictionary/assets/137730219/392edaf0-def9-43fd-8c8b-8e62256384bc)
That's what it says in my code. I didn't change anything.